### PR TITLE
Add review checklist

### PR DESCRIPTION
**Pull request by maltekliemann**
This list is based on a mixture of personal pet peeves and things that I have trouble remembering or found surprising. Feel free to add your own in the suggestions!

Also, we may want to think about using this as a PR template. Especially now that bounties are being set up this might be helpful to new contributors.

### CODE REVIEW

File: docs/review_checklist.md
Comment (maltekliemann) [reply docs/review_checklist.md:31]: Feel free to raise an issue about this. Might be interesting to have this, although I think this is a difficult thing to solve. For example, will these tools still allow us to use substrate calls, which often contain panickers?

File: docs/review_checklist.md
Comment (maltekliemann) [reply docs/review_checklist.md]: 98ee686664182f188383546f6d0f093de65df68b

File: docs/review_checklist.md
Comment (maltekliemann) [reply docs/review_checklist.md]: 98ee686664182f188383546f6d0f093de65df68b

File: docs/review_checklist.md
Comment (maltekliemann) [reply docs/review_checklist.md]: edd3b1c17321a1a92727bf92694c5a8c4dc4a4c4

Comment (maltekliemann) [reply docs/review_checklist.md]: (Not all compiler warnings seem to be caught by the CI.)

File: docs/review_checklist.md
Comment (maltekliemann) [reply docs/review_checklist.md]: 14b8bfc250664077b782b9e66d133825a9372add

File: docs/review_checklist.md
Comment (maltekliemann) [reply docs/review_checklist.md]: Not exactly. By "local node" I mean the local temporary dev node. Checking that the node syncs with the live networks is a different thing. I've added commands for each in 14b8bfc250664077b782b9e66d133825a9372add.

**Review by Chralt98 — APPROVED** [review]

File: docs/review_checklist.md
Comment (sea212) [inline docs/review_checklist.md:39]:
```suggestion
- [ ] Code quality:
  - [ ] Avoidable compiler warnings were resolved.
  - [ ] Integer arithmetic is only saturated/checked and all panickers are removed.
  - [ ] Code contained in macro invocations (benchmarks,
        `runtime/common/src/lib.rs`, `decl_runtime_apis!`) is correctly formatted.
  - [ ] All `*.toml` files are formatted with `taplo` (run
        `taplo format --check`).
  - [ ] All copyright notices are up to date.
  - [ ] Enums are sorted alphabetically, except for enums used in storage (to
        prevent migrations), errors and events.
```

File: docs/review_checklist.md
Comment (sea212) [inline docs/review_checklist.md:45]:
```suggestion
- [ ] In case configuration items or storage elements were changed: Necessary storage migrations are provided.
```

File: docs/review_checklist.md
Comment (sea212) [inline docs/review_checklist.md:47]: Adjusted two things here beside the conversion to an action:
1. Removed the constraint "mainnet only", because changing values can also lead to undesired issues on the testnet
2. Replaced project owners by code owners.

```suggestion
- [ ] In case configuration values changed: The
      implications have been discussed with the [code owners](https://github.com/zeitgeistpm/zeitgeist/blob/main/CODEOWNERS).
```

File: docs/review_checklist.md
Comment (sea212) [inline docs/review_checklist.md:55]:
```suggestion
- [ ] If the changes include a storage migration:
  - [ ] The affected pallet's `STORAGE_VERSION` was bumped.
  - [ ] Try-runtime checks were added and the following conditions were ensured:
    - [ ] The storage migration bumps the pallet version correctly.
    - [ ] The try-runtime _fails_ if you comment out the migration code.
    - [ ] The try-runtime passes without any warnings (substrate storage
          operations often just log a warning instead of failing, so these
          warnings usually point to problem which could break the storage).
```

File: docs/review_checklist.md
Comment (sea212) [inline docs/review_checklist.md:59]:
```suggestion
_All_ modifications of the on-chain storage **must** be broadcast to indexers
```

File: docs/review_checklist.md
Comment (sea212) [inline docs/review_checklist.md:62]:
```suggestion
Zeitgeist's business logic builds on, for example pallet-balances. Examples of
```

**Review by sea212 — COMMENTED** [review]
Thanks for creating this extensive list.

The review mainly suggests two changes:
1. Instead of (primarily) using questions in the checklist, actions should be used. A checklist is a list of actions that can be checked off once done (i.e. send e-mail to xy).
2. Replaced all personal pronouns

**Comment by sea212** [comment]
What do you think about adding a workflow that automatically adds this checklist in a PR that is marked as ready for review? That way we can collaboratively just check off what has been verified.
This is one example how to achieve that: https://stackoverflow.com/a/74790862/10273801

**Review by Chralt98 — APPROVED** [review]

**Review by sea212 — APPROVED** [review]

**Comment by maltekliemann** [comment]
> What do you think about adding a workflow that automatically adds this checklist in a PR that is marked as ready for review?

I worry this might be overkill. If somebody changes two lines in prediction-markets, do we really go through that entire list? It's meant as a collection of things to watch out for, but I don't think I want to mess with the way that each of us does our reviews.

**Comment by sea212** [comment]
@maltekliemann Can we merge this?

**Comment by maltekliemann** [comment]
> @maltekliemann Can we merge this?

Yeah, was just waiting for the release to be published.

**Pull request merged**
